common/src/parsing.rs

```diff
@@ -21,22 +21,53 @@ where
     RhsMissing,
 }
 
-/// Parse a range from a string.
+/// Parse an exclusive range from a string.
 ///
 /// A valid range is of the form `lhs..rhs`, where `lhs` and `rhs` are numbers.
 ///
 /// # Example
 ///
 /// ```rust
-/// # use common::parsing::parse_range;
-/// assert_eq!(parse_range::<usize>("0..10"), Ok(0..10));
+/// # use common::parsing::parse_range_exclusive;
+/// assert_eq!(parse_range_exclusive::<usize>("0..10"), Ok(0..10));
 /// ```
-pub fn parse_range<T>(s: &str) -> Result<Range<T>, RangeParseError<T>>
+pub fn parse_range_exclusive<NumberT>(s: &str) -> Result<Range<NumberT>, RangeParseError<NumberT>>
```

Comment:
nit: not need to have it public I believe
```suggestion
pub(crate) fn parse_range_exclusive<NumberT>(s: &str) -> Result<Range<NumberT>, RangeParseError<NumberT>>
```

Reply:
Doc tests fail with this function being  pub(crate)

Reply:
I don't think that's a good reason to keep a function public

Reply:
It is a common non stateless function, I don't see a reason to force it to be private

Reply:
I'd tend to prefer keeping private / crate-restricted everything that _doesn't need_ to be made public, and expose only what's necessary. Curious about other opinions though.

Reply:
Ok, well remove then the doc tests and make it private